animations/src/main/java/com/chiralbehaviors/CoRE/meta/workspace/dsl/WorkspaceImporter.java

```diff
@@ -167,32 +167,50 @@ public WorkspaceImporter(WorkspacePresentation wsp, Model model) {
         this.em = model.getEntityManager();
     }
 
-    public WorkspaceAccessor addToWorkspace() {
-        scope = model.getWorkspaceModel()
-                     .getScoped(getWorkspaceProduct());
-        workspace = (EditableWorkspace) scope.getWorkspace();
-        loadWorkspace();
+    public WorkspaceAccessor getWorkspace() {
         return workspace;
     }
 
-    public WorkspaceAccessor createWorkspace() {
-        scope = model.getWorkspaceModel()
-                     .createWorkspace(createWorkspaceProduct(),
-                                      model.getCurrentPrincipal()
-                                           .getPrincipal());
-        workspace = (EditableWorkspace) scope.getWorkspace();
-        loadWorkspace();
-        return workspace;
+    public void initialize() {
+        workspaceUri = stripQuotes(wsp.getWorkspaceDefinition().uri.getText());
+        uuid = WorkspaceAccessor.uuidOf(workspaceUri);
     }
 
-    public WorkspaceAccessor getWorkspace() {
-        return workspace;
+    public WorkspaceAccessor manifest() {
+        initialize();
+        int version = Integer.parseInt(wsp.getWorkspaceDefinition().version.getText());
+        return version == 1 ? createWorkspace() : addToWorkspace(version);
     }
 
     public void setScope(WorkspaceScope scope) {
         this.scope = scope;
     }
 
+    private WorkspaceAccessor addToWorkspace(int version) {
+        Product definingProduct = getWorkspaceProduct();
+        if (definingProduct == null) {
+            throw new IllegalStateException(String.format("Workspace %s does not exist, cannot update to version %s",
+                                                          wsp.getWorkspaceDefinition().name.getText(),
+                                                          version));
+        }
+        if (definingProduct.getVersion() != version - 1) {
```

Comment:
I have an [issue](https://github.com/ChiralBehaviors/Ultrastructure/issues/135) for differential loading.  Certainly, the model is "you can't skip a version" and the version is a single integer.